redaxo/src/addons/media_manager/lib/effects/effect_convert2img.php

```diff
@@ -44,6 +53,41 @@ public function execute()
             $convertTo = self::CONVERT_TO[(string) $this->params['convert_to']];
         }
 
+        if ($this->isVideoToImageConversionSupported()) {
+
+            $inputFile = $this->media->getMediaPath();
+
+            // Try to get the duration of the video using ffprobe
+            $ffprobeCmd = 'ffprobe -v error -show_entries format=duration -of default=noprint_wrappers=1:nokey=1 ' . escapeshellarg($inputFile);
+            $duration = exec($ffprobeCmd);
+
+            if ($duration) {
+                $timestamp = gmdate('H:i:s', (int) floor($duration / 2));
+            } else {
+                $timestamp = '00:00:01';
+            }
+
+            $outputFile = rex_path::addonCache('media_manager', 'media_manager__convert2img_' . md5($inputFile) . '.' . $convertTo['ext']);
+
+            $cmd = 'ffmpeg -y -i ' . escapeshellarg($inputFile) . ' -ss ' . escapeshellarg($timestamp) . ' -vframes 1 ' . escapeshellarg($outputFile);
+            exec($cmd, $out, $ret);
+
+            if (0 !== $ret) {
+                throw new rex_exception('Unable to exec command ' . $cmd);
+            }
+
+            $this->media->setSourcePath($outputFile);
+            $this->media->refreshImageDimensions();
+            $this->media->setFormat($convertTo['ext']);
+            $this->media->setHeader('Content-Type', $convertTo['content-type']);
+            $filename = $this->media->getMediaFilename();
+            $this->media->setMediaFilename($filename);
```

Comment:
Sollte hier nicht besser auch die Extension des Media-Filenames geändert werden?
Der Filename wird ja im `Content-Disposition`-Header ausgeliefert, und ist also relevant, wenn jemand das Bild herunterlädt.

Reply:
Der image_format-Effekt macht das (der Bilder in webp/avif umwandelt).
Dachte hätten es hier in convert2img auch drin, ist aber anscheinend nicht so.
Daher ok, wenn du es bei den Videos dann auch weglässt.